watertap/unit_models/tests/test_ion_exchange_0D.py

```diff
@@ -10,1070 +10,287 @@
-            "bed_expansion_frac_B",
-            "bed_expansion_frac_C",
-            "bw_rate",
-            "distributor_h",
-            "number_columns_redund",
-            "p_drop_A",
-            "p_drop_B",
-            "p_drop_C",
-            "pump_efficiency",
-            "rinse_bv",
-            "service_to_regen_flow_ratio",
-            "t_bw",
-            "t_regen",
-            "underdrain_h",
-        ]
-
-        for p in ix_params:
-            assert hasattr(ix, p)
-            param = getattr(ix, p)
-            assert isinstance(param, Param)
-
-        ix_vars = [
-            "N_Pe_bed",
-            "N_Pe_particle",
-            "N_Re",
-            "N_Sc",
-            "N_Sh",
-            "bed_depth",
-            "bed_porosity",
-            "bed_vol_tot",
-            "c_norm",
-            "col_diam",
-            "col_height",
-            "col_height_to_diam_ratio",
-            "dimensionless_time",
-            "ebct",
-            "fluid_mass_transfer_coeff",
-            "langmuir",
-            "mass_removed",
-            "num_transfer_units",
-            "number_columns",
-            "partition_ratio",
-            "resin_bulk_dens",
-            "resin_diam",
-            "resin_eq_capacity",
-            "resin_max_capacity",
-            "resin_surf_per_vol",
-            "resin_unused_capacity",
-            "service_flow_rate",
-            "t_breakthru",
-            "vel_bed",
-        ]
-
-        for v in ix_vars:
-            assert hasattr(ix, v)
-            var = getattr(ix, v)
-            assert isinstance(var, Var)
-
-        # test statistics
-        assert number_variables(m) == 70
-        assert number_total_constraints(m) == 44
-        assert number_unused_variables(m) == 10
-
-    @pytest.mark.unit
-    def test_dof(self, IX_lang):
-        m = IX_lang
-        check_dof(m, fail_flag=True)
-
-    @pytest.mark.unit
-    def test_calculate_scaling(self, IX_lang):
-        m = IX_lang
-        m.fs.properties.set_default_scaling(
-            "flow_mol_phase_comp", 1e-4, index=("Liq", "H2O")
-        )
-        m.fs.properties.set_default_scaling(
-            "flow_mol_phase_comp", 10, index=("Liq", "Ca_2+")
-        )
-
-        calculate_scaling_factors(m)
-
-        # check that all variables have scaling factors
-        unscaled_var_list = list(unscaled_variables_generator(m))
-        assert len(unscaled_var_list) == 0
-
-    @pytest.mark.component
-    def test_initialize(self, IX_lang):
-        m = IX_lang
-        initialization_tester(m, unit=m.fs.ix, outlvl=idaeslog.DEBUG)
-
-    @pytest.mark.component
-    def test_solve(self, IX_lang):
-        m = IX_lang
-        results = solver.solve(m, tee=True)
-        assert_units_consistent(m)
-        # Check for optimal solution
-        assert_optimal_termination(results)
-
-    @pytest.mark.requires_idaes_solver
-    @pytest.mark.component
-    def test_mass_balance(self, IX_lang):
-        m = IX_lang
```

Comment:
We should try to keep at least one of these conservation checks in the testing. See #1372 for examples

Reply:
Yes agreed, but wasn't sure how to do so with the test harness without just keeping the existing test, thanks for the direction

Reply:
Eventually conservation checks will work their way into the test harness, but it's not a high priority atm.